lib/cn/cn_tree.c

```diff
@@ -1353,71 +1327,99 @@ cn_comp_update_kvcompact(struct cn_compaction_work *work, struct kvset *new_kvse
     }
 }
 
-/**
- * cn_comp_update_spill() - update tree after spill operation
- * See section comment for more info.
- */
+static bool
+cn_node_wait_for_turn(struct cn_compaction_work *w, struct cn_tree_node *tn)
+{
+    atomic_long *node_sgen = &w->cw_node->tn_sgen;
+    struct kvdb_health *hp = w->cw_tree->ct_kvdb_health;
+    bool ready;
+
+    if (w->cw_err)
+        return false;
+
+    while (w->cw_sgen != atomic_read(node_sgen) + 1 &&
+           !atomic_read(w->cw_cancel_request) &&
+           !kvdb_health_check(hp, KVDB_HEALTH_FLAG_ALL)) {
+
+        usleep(100);
+    }
+
+    ready = !atomic_read(w->cw_cancel_request) && !kvdb_health_check(hp, KVDB_HEALTH_FLAG_ALL);
+
+    return ready;
+}
```

Comment:
Seems like it could be waiting a long time, right?  Better to use a condvar here vs effectively spinning...

Reply:
Yes, that'll be better. Let me fix it.